# 전국 신규 민간 아파트 분양가격 분석하기

## 공공데이터 다운로드 후 로드하기

### 데이터셋 다운로드 받기

http://bit.ly/open-data-set-folder

* 전국 평균 분양가격(2013~2015)
* 전국 평균 분양가격(~2019)           

분양가격은 1,000원 단위임.

### 판다스 import 하기

In [1]:
import pandas as pd

### 파일을 로드해서 변수에 담기

In [2]:
# pd.read_csv("data/주택도시보증공사_전국 평균 분양가격(2019년 12월).csv")
# UnicodeDecodeError가 발생. 기본은 utf-8, 한국 전산시스템이 한국 완성형 코드인 'EUC-KR'을 사용하기 때문.(2350자만 표현가능)
df_last = pd.read_csv("data/주택도시보증공사_전국 평균 분양가격(2019년 12월).csv",encoding="euc-kr") #cp949로 지정해도 가능

In [3]:
df_last.shape #(행, 열)

(4335, 5)

In [4]:
#데이터 미리보기
df_last.head() #shift + tab을 누르면 메서드의 도움말을 볼 수 있다.

지역명               규모구분    연도   월 분양가격(㎡)
0  서울                 전체  2015  10    5841
1  서울         전용면적 60㎡이하  2015  10    5652
2  서울   전용면적 60㎡초과 85㎡이하  2015  10    5882
3  서울  전용면적 85㎡초과 102㎡이하  2015  10    5721
4  서울        전용면적 102㎡초과  2015  10    5879

In [5]:
df_last.tail()

지역명               규모구분    연도   월 분양가격(㎡)
4330  제주                 전체  2019  12    3882
4331  제주         전용면적 60㎡이하  2019  12     NaN
4332  제주   전용면적 60㎡초과 85㎡이하  2019  12    3898
4333  제주  전용면적 85㎡초과 102㎡이하  2019  12     NaN
4334  제주        전용면적 102㎡초과  2019  12    3601

In [6]:
df_first = pd.read_csv("data/전국 평균 분양가격(2013년 9월부터 2015년 8월까지).csv",encoding = "cp949")

In [7]:
df_first.shape

(17, 22)

In [8]:
df_first.head()

지역  2013년12월  2014년1월  2014년2월  2014년3월  2014년4월  2014년5월  2014년6월  \
0  서울     18189    17925    17925    18016    18098    19446    18867   
1  부산      8111     8111     9078     8965     9402     9501     9453   
2  대구      8080     8080     8077     8101     8267     8274     8360   
3  인천     10204    10204    10408    10408    10000     9844    10058   
4  광주      6098     7326     7611     7346     7346     7523     7659   

   2014년7월  2014년8월  ...  2014년11월  2014년12월  2015년1월  2015년2월  2015년3월  \
0    18742    19274  ...     20242     20269    20670    20670    19415   
1     9457     9411  ...      9208      9208     9204     9235     9279   
2     8360     8370  ...      8439      8253     8327     8416     8441   
3     9974     9973  ...     10020     10020    10017     9876     9876   
4     7612     7622  ...      7752      7748     7752     7756     7861   

   2015년4월  2015년5월  2015년6월  2015년7월  2015년8월  
0    18842    18367    18374    18152    18443  
1     9327     9345     9515     9559     9581  
2     8446     8568     8542     8542     8795  
3     9938    10551    10443    10443    10449  
4     7914     7877     7881     8089     8231  

[5 rows x 22 columns]

In [9]:
df_first.tail()

지역  2013년12월  2014년1월  2014년2월  2014년3월  2014년4월  2014년5월  2014년6월  \
12  전북      6282     6281     5946     5966     6277     6306     6351   
13  전남      5678     5678     5678     5696     5736     5656     5609   
14  경북      6168     6168     6234     6317     6412     6409     6554   
15  경남      6473     6485     6502     6610     6599     6610     6615   
16  제주      7674     7900     7900     7900     7900     7900     7914   

    2014년7월  2014년8월  ...  2014년11월  2014년12월  2015년1월  2015년2월  2015년3월  \
12     6319     6436  ...      6583      6583     6583     6583     6542   
13     5780     5685  ...      5768      5784     5784     5833     5825   
14     6556     6563  ...      6881      6989     6992     6953     6997   
15     6613     6606  ...      7125      7332     7592     7588     7668   
16     7914     7914  ...      7724      7739     7739     7739     7826   

    2015년4월  2015년5월  2015년6월  2015년7월  2015년8월  
12     6551     6556     6601     6750     6580  
13     5940     6050     6243     6286     6289  
14     7006     6966     6887     7035     7037  
15     7683     7717     7715     7723     7665  
16     7285     7285     7343     7343     7343  

[5 rows x 22 columns]

## 데이터 요약하기

In [10]:
df_last.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4335 entries, 0 to 4334
Data columns (total 5 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   지역명      4335 non-null   object
 1   규모구분     4335 non-null   object
 2   연도       4335 non-null   int64 
 3   월        4335 non-null   int64 
 4   분양가격(㎡)  4058 non-null   object
dtypes: int64(2), object(3)
memory usage: 169.5+ KB


분양가격은 결측치 때문에 개수가 다르다.

### 결측치 보기

In [11]:
True == 1

True

In [12]:
False == 0

True

In [13]:
True + True + False

2

In [14]:
df_last.isnull()

지역명   규모구분     연도      월  분양가격(㎡)
0     False  False  False  False    False
1     False  False  False  False    False
2     False  False  False  False    False
3     False  False  False  False    False
4     False  False  False  False    False
...     ...    ...    ...    ...      ...
4330  False  False  False  False    False
4331  False  False  False  False     True
4332  False  False  False  False    False
4333  False  False  False  False     True
4334  False  False  False  False    False

[4335 rows x 5 columns]

In [15]:
df_last.isnull().sum()

지역명          0
규모구분         0
연도           0
월            0
분양가격(㎡)    277
dtype: int64

In [16]:
df_last.isna().sum() # innull()과 결과는 동일

지역명          0
규모구분         0
연도           0
월            0
분양가격(㎡)    277
dtype: int64

 ### 데이터 타입 변경
 분양가격이 object(문자) 타입으로 되어있다. 문자열 타입은 계산을 할 수 없기에(sum, avg값 등) 수치 데이터로 변경해준다. 그런데 결측치가 섞여 있으면 변환이 제대로 되지 않는다. 그래서 pd.to_numeric을 사용하여 데이터의 타입을 변경해주어야 한다.

In [17]:
df_last["분양가격(㎡)"] #분양가격 열만 보기

0       5841
1       5652
2       5882
3       5721
4       5879
        ... 
4330    3882
4331     NaN
4332    3898
4333     NaN
4334    3601
Name: 분양가격(㎡), Length: 4335, dtype: object

In [18]:
# df_last["분양가격(㎡)"].astype(int) -> 결측치 때문에 ValueError가 발생

In [19]:
# pd.to_numeric(df_last["분양가격(㎡)"]) -> 에러발생

In [20]:
pd.to_numeric(df_last["분양가격(㎡)"], errors='coerce')

0       5841.0
1       5652.0
2       5882.0
3       5721.0
4       5879.0
         ...  
4330    3882.0
4331       NaN
4332    3898.0
4333       NaN
4334    3601.0
Name: 분양가격(㎡), Length: 4335, dtype: float64

왜 float로 나오는가?

In [21]:
type(pd.np.nan) #pandas에 내장된 numpy의 nan이 float형이기 떄문에

C:\Users\Study\AppData\Local\Temp/ipykernel_14044/3367481099.py:1: FutureWarning: The pandas.np module is deprecated and will be removed from pandas in a future version. Import numpy directly instead
  type(pd.np.nan) #pandas에 내장된 numpy의 nan이 float형이기 떄문에


float

In [22]:
df_last["분양가격"] = pd.to_numeric(df_last["분양가격(㎡)"], errors='coerce')

In [23]:
df_last["분양가격"].mean() #평균

3238.128632802628

In [24]:
df_last["분양가격"].sum() #합

12813275.0

In [25]:
df_last["분양가격"].max() #최댓값

12728.0

### 평당분양가격 구하기
2013년부터의 데이터는 평당분양가격 기준인데 반해, 최근데이터는 1제곱미터당 평균분양가격이기 때문에 단위를 똑같이 만들어줄 필요가 있다.

In [26]:
df_last["분양가격"] * 3.3

0       19275.3
1       18651.6
2       19410.6
3       18879.3
4       19400.7
         ...   
4330    12810.6
4331        NaN
4332    12863.4
4333        NaN
4334    11883.3
Name: 분양가격, Length: 4335, dtype: float64

In [27]:
df_last["평당분양가격"] = df_last["분양가격"] * 3.3
df_last

지역명               규모구분    연도   월 분양가격(㎡)    분양가격   평당분양가격
0     서울                 전체  2015  10    5841  5841.0  19275.3
1     서울         전용면적 60㎡이하  2015  10    5652  5652.0  18651.6
2     서울   전용면적 60㎡초과 85㎡이하  2015  10    5882  5882.0  19410.6
3     서울  전용면적 85㎡초과 102㎡이하  2015  10    5721  5721.0  18879.3
4     서울        전용면적 102㎡초과  2015  10    5879  5879.0  19400.7
...   ..                ...   ...  ..     ...     ...      ...
4330  제주                 전체  2019  12    3882  3882.0  12810.6
4331  제주         전용면적 60㎡이하  2019  12     NaN     NaN      NaN
4332  제주   전용면적 60㎡초과 85㎡이하  2019  12    3898  3898.0  12863.4
4333  제주  전용면적 85㎡초과 102㎡이하  2019  12     NaN     NaN      NaN
4334  제주        전용면적 102㎡초과  2019  12    3601  3601.0  11883.3

[4335 rows x 7 columns]

### 분양가격 요약하기 .describe()

In [28]:
df_last.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4335 entries, 0 to 4334
Data columns (total 7 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   지역명      4335 non-null   object 
 1   규모구분     4335 non-null   object 
 2   연도       4335 non-null   int64  
 3   월        4335 non-null   int64  
 4   분양가격(㎡)  4058 non-null   object 
 5   분양가격     3957 non-null   float64
 6   평당분양가격   3957 non-null   float64
dtypes: float64(2), int64(2), object(3)
memory usage: 237.2+ KB


In [30]:
# 변경 전 컬럼인 분양가격(㎡) 요약. 
df_last["분양가격(㎡)"].describe()

count     4058
unique    1753
top       2221
freq        17
Name: 분양가격(㎡), dtype: object

- unique : 고유한 데이터
- top : 가장 빈번하게 등장하는 데이터 -> 2221
- freq : 가장 빈번하게 등장한 데이터가 등장한 횟수 -> 2221이 17번 등장했다.

In [33]:
# 수치데이터로 변경된 분양가격 요약.
df_last["분양가격"].describe() # float이기 떄문에 요약 결과가 다르게 나온다.

count     3957.000000
mean      3238.128633
std       1264.309933
min       1868.000000
25%       2441.000000
50%       2874.000000
75%       3561.000000
max      12728.000000
Name: 분양가격, dtype: float64

왜 float에서의 전체 데이터 개수(count)가 더 작은가?
-> object에서는 공백 데이터도 결측치가 아니라 데이터가 있는 것으로 간주

- mean : 평균값
- std : 표준편차
- 25% : 제 1사분위수 -> 데이터의 25%가 이 값보다 작거나 같음.
- 50% : 제 2사분위수 -> 데이터의 50%가 이 값보다 작거나 같음.
- 75% : 제 3사분위수 -> 데이터의 75가 이 값보다 작거나 같음.

 **.describe()** 에서 중요한 것은 **평균값(mean)과 중앙값(50%)의 차이**
 
 중앙값보다 평균값이 훨씬 높다. -> 최댓값 때문에 평균값이 더 높게 나타났다.